src/core/pythoncapi_compat.h

```diff
@@ -0,0 +1,393 @@
+// Header file providing new functions of the Python C API to old Python
```

Comment:
May I ask you to move this file to `src/core/lib/pythoncapi`? We normally keep all the third party libraries there.

Reply:
Oh sure: done in my second commit. Tell me if you prefer that I squash my commits of this PR.

Reply:
There is no need to squash. Thanks.